File: be/src/main/java/codesquad/secondhand/dto/category/CategoryDto.java
package codesquad.secondhand.dto.category;

import codesquad.secondhand.entity.Category;
import lombok.Getter;
import lombok.RequiredArgsConstructor;

@Getter
@RequiredArgsConstructor
public class CategoryDto {

    private final Long idx;
    private final String name;
    private final String imgUrl;

    public static CategoryDto of(Category category) {
        return new CategoryDto(category.getCategoryIdx(),
                category.getName(),
                category.getImageUrl());
    }

}

File: iOS/SecondHand/SecondHand/Category/CategoryDTO.swift
//
//  CategoryDTO.swift
//  SecondHand
//
//  Created by Wood on 2023/08/14.
//

import Foundation

struct CategoryDTO: DTO, Codable {
    struct CategoryData: Codable {
        var idx: Int
        var name: String
        var imgUrl: String
    }
    
    var success: Bool
    var status: Int
    var code: Int
    var message: String
    var data: [CategoryData]
}
